### PR TITLE
Install .desktop and icon files

### PR DESCRIPTION
**Pull request by masenf**
For linux platforms, install a .desktop file and icon for launching CHiRP.

This works for both user-installs (~/.local) and system-installs. Uninstalling via `pip uninstall` properly removes the files.

--------

Allow `python3 -m chirp.wxui`, to start chirp without invoking the console script `chirp`. If the user hasn't added `~/.local/bin` to their PATH, then the desktop launcher can't easily launch the script. This avoids having to set a different Exec= line in the desktop file for user vs. system install (.local/bin/chirp vs chirp)

# CHIRP PR Checklist

The following must be true before PRs can be merged:

* All tests must be passing.
* Commits should be squashed into logical units.
* Commits should be rebased (or simply rebase-able in the web UI) on current master. Do not put merge commits in a PR.
* Commits in a single PR should be related.
* Major new features or bug fixes should reference a [CHIRP issue](https://chirp.danplanet.com/projects/chirp/issues).
* New drivers should be accompanied by a test image in `tests/images` (except for thin aliases where the driver is sufficiently tested already).

Please also follow these guidelines:

* Keep cleanups in separate commits from functional changes.
* Please write a reasonable commit message, especially if making some change that isn't totally obvious (such as adding a new model, adding a feature, etc).
* Do not add new py2-compatibility code (No new uses of `six`, `future`, etc).
* All new drivers should set `NEEDS_COMPAT_SERIAL=False` and use `MemoryMapBytes`.
* New drivers and radio models will affect the Python3 test matrix. You should regenerate this file with `tox -emakesupported` and include it in your commit.

### CODE REVIEW

**Comment by kk7ds** [comment]
CC @vishwin

File: chirp/wxui/__main__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""Entry point for starting chirp via `python -m chirp.wxui`"""
+from . import chirpmain
+
+chirpmain()
```
Comment (kk7ds) [inline chirp/wxui/__main__.py:4]: I don't understand why we need to do this (meaning you running it with `python3 -mchirp.wxui` in the desktop file) when we have `chirp` installed via `console_scripts`. Can't you just run `chirp`?

Comment (kk7ds) [reply chirp/wxui/__main__.py:4]: Okay makes sense. Annoying, but fair enough :)

I'mma wait for @vishwin to look over this as he was pushing the `setup.py` to the current state.

**Comment by vishwin** [comment]
Apart from `chirp.wxui.__main__`, all of this is not necessary and actually wrong, as [`data_files` is deprecated](https://github.com/pypa/setuptools/discussions/2648).

Within the Python packaging ecosystem, there is no reliable way to install and manage platform-specific files and locations like `.desktop`, hence why they are located within the package and left as further exercises (for system package managers, etc). `entry_points` is the lone exception as it can be reliably managed across platforms. Certain users not having `~/.local/bin` in `$PATH` is not our problem, but having `chirp.wxui` as an executable module helps with providing entry points, especially when not installed as a (Python) package.

As for current state of `setup.py`, it is officially temporary and a stop-gap until we implement PEP-517 support.

**Comment by masenf** [comment]
Thanks for looking. Will leave the .desktop as documentation then and submit separate PR for the main entry point.

**Comment by kk7ds** [comment]
@masenf How about a command-line option to install a desktop file, which we only expose on linux? Then once you've got it installed, you can:
```
$ python3 -mchirp.wxui --install-desktop
```
once and it'll just barf the desktop file into the user or system directory depending on if the system location is writeble or not?

**Pull request closed without merging**